runtime/src/accounts_index.rs

```diff
@@ -787,6 +838,23 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
                 ongoing_scan_roots.remove(&max_root);
             }
         }
+
+        // If the fork with tip at bank `scan_bank_id` was removed durin our scan, then the scan
```

Comment:
The check before the scan operation is to avoid incurring the cost of the scan if we already know that the slot has been dumped

The check after the scan operation is to account for slot dumping that occurred during the scan. Unfortunately, it's as you said, if the dump happens after the scan was finished and before this check is made, then we chalk this up to bad luck and still abort the results. I don't know of a great way to avoid this without blocking the remove during an ongoing scan.

Reply:
If the latter case can happen, which is after the scan and before the check, then it is also possible it can happen after we did the check and receive the negative answer and then it happens right after that while we already made decision to return a success. Would that have any correctness concern?

Reply:
@lijunwangs correct that can happen as well. That should be ok because the results returned in that case are guaranteed to not have been corrupted by the slot dumping, which upholds our guarantee that scans should be consistent up to slot *boundaries*